unittests/Basic/DiagnosticTest.cpp

```diff
@@ -6,6 +6,7 @@
 //
 //===----------------------------------------------------------------------===//
 
+#define private public
 #include "clang/Basic/Diagnostic.h"
```

Comment:
You can probably do `#undef private` here to demonstrate where we need to access private information.